cdap-app-fabric/src/main/java/co/cask/cdap/security/DefaultUGIProvider.java

```diff
@@ -87,6 +88,8 @@ public UserGroupInformation getConfiguredUGI(ImpersonationInfo impersonationInfo
       String expandedPrincipal = SecurityUtil.expandPrincipal(impersonationInfo.getPrincipal());
       LOG.debug("Logging in as: principal={}, keytab={}", expandedPrincipal, localKeytabFile);
 
+      FileUtils.assertReadableFile(localKeytabFile);
```

Comment:
Why need a new util class? Isn't that you just need to check for `Files.isReadable`?